host_camera_service/host_camera_server.cc

```diff
@@ -262,24 +265,16 @@ int open_camera()
 
 void* InitCamera(void *arg)
 {
-    atomic<bool> request_negotiation = false;
     while(true) {
-        while (!video_sink->IsConnected() || !request_negotiation) {
-                request_negotiation = false;
-
-            if (!request_negotiation && video_sink->IsConnected()) {
-		    cout <<"[Stream] now its connected";
-                video_sink->GetCameraCapabilty();
-                std::vector<VideoSink::camera_info_t> camera_info(NUM_OF_CAMERAS_REQUESTED);
-                for (int i = 0; i < NUM_OF_CAMERAS_REQUESTED; i++) {
-                    camera_info[i].codec_type = (VideoSink::VideoCodecType)v4l2_format;
-                    camera_info[i].resolution = VideoSink::FrameResolution::k1080p;
-                }
-                                video_sink->SetCameraCapabilty(camera_info);
-                request_negotiation = true;
-	    }
-	}
-	usleep(10000);
+        video_sink->ResetCameraCapabilty();
```

Comment:
I think you are trying to wait for the vhal connection in the ResetCameraCapabilty.
If so, shouldn't the API named differently. Eg. WaitForConnect ? Also, that API return should be triggered based on the connection status of socket_client_ .